docs/pages/setup/security/reduce-blast-radius.mdx

```diff
@@ -0,0 +1,276 @@
+
+As an illustration, we have assigned user `myuser` to the `user` role, which we defined using the following template:
+
+```yaml
+kind: role
+version: v4
+metadata:
+  name: user
+spec:
+  deny:
+    request:
+      roles: ['admin']
+```
+
+Next, `myuser` attempts to request the `admin` role.
+
+```code
+$ tsh request create --roles=admin
+```
+
+However, the Auth Service denies the request.
+
+```code
+Creating request...
+ERROR: user "myuser" can not request role "admin"
+```
+
+
+## Restrict role requests based on user traits
+Teleport's `role` resource lets you take precautions against accidental privilege escalation by ensuring that any user with particular attributes will have restricted access to certain roles. You can assign a list of `traits` to a user, then define a `role` resource that prevents any user whose traits match a regular expression from requesting elevated privileges.
+
+A user has the same traits regardless of the roles they acquire. As a result, if a user happens to obtain another role because of an RBAC oversight, you can use trait-based restrictions to stop them from requesting a role with even more privileges.
+
+Let's say that you have defined the following role for a contractor you have hired to analyze financial data.
+
+```yaml
+kind: user
+version: v2
+metadata:
+  name: myuser
+spec:
+  roles:
+    - analyst # An unprivileged role
+  traits:
+    logins:
+      - myuser
+    groups:
+      - contractors
+```
+
+Analysts sometimes need write access to your organization's database in order to create stored procedures, and can request access to the `db-writer` role. Only trusted analysts can request this access, and belong to a special `admins` group. Using `deny` rules, you can prevent analysts who are _not_ in the `admins` group from requesting access to the `db-writer` role:
+
+```yaml
+kind: role
+version: v4
+metadata:
+  name: analyst
+spec:
+  deny:
+    request:
+      claims_to_roles:
+        - claim: groups
+          value: "{{regexp.not_match(\"admin\")}}"
+          roles: ["db-writer"]
+  allow:
+    request:
+      roles: ["db-writer"]
+      thresholds:
+        - approve: 2
+          deny: 1
+```
+
+The `claims_to_roles` field within an `allow` or `deny` rule maps a user's `traits` to `roles` that they are either permitted or forbidden to request. In this case, we use the `{{regexp.not_match(\"admin\")}}` template function to prevent any user  from requesting the `db-writer` role unless they have a `groups` trait with a value like `administrator` or `admins`. Users who _do_ have such a trait can request the role with two approvals.
+
+
+## Set up your RBAC without admin roles
+You can design your Teleport RBAC so that there is no all-powerful administrator in the system, or even a `reviewer` role with elevated privileges. This way, you can reduce the blast radius if an attacker successfully impersonates a Teleport user and requests a more privileged role.
```

Comment:
Bit of a silly nitpick, but I feel like the example used in this section undermines the point being made here a bit.  The example below shows a user role that can only be partially escalated to editor, which seems to imply the existence of an admin-like actor with full privileges (the thing we're saying you can choose to avoid).

I think having separate examples for the "democratic" and "partial escalation" scenarios would make the points clearer and more memorable.

Reply:
@fspmarshall @ptgott with access requests you can have 0 admins in the system at any given time. However when someone gets the privilege, they would become admin temporarily. We can limit what they can admin exactly, but that is only partial mitigation